### ISSUE TITLE
[Q][Feature]: Handling of relay server allocation

### ISSUE THREAD

**Issue by ltguillaume** [simplex-chat/simplex-chat#3712]
### Question

When creating new connections, as is documented, random relay servers are allocated. It seems like this does NOT take into account the current user's location, which could lead to 1) very high latencies and 2) very low bandwidth. Furthermore, I'd prefer my devices not to connect to servers situated in countries that don't take privacy seriously.

1. As documented, the endpoints for the subdomains of the XMP servers may change at any time, so checking the IPs of every endpoint and then disabling the ones I don't want to use wouldn't make sense. And that's just on my side of the connection.
2. If I would set up my own relay server, I cannot easily make my friends use that server exclusively. To do such a thing, there would have to be a way to __export XMP/XFTP server configuration__ to a URI, which they could import.

**Comment by Sepero** [comment]
> When creating new connections, as is documented, random relay servers are allocated. It seems like this does NOT take into account the current user's location, which could lead to 1) very high latencies and 2) very low bandwidth.

If you only used certain servers for communications, then wouldn't that make your communications less anonymous?

> Furthermore, I'd prefer my devices not to connect to servers situated in countries that don't take privacy seriously.

Since all messages are E2EE, then you seem to want to hide your Metadata from certain countries. Which countries do you think don't collect Metadata?

**Comment by 2Belette** [comment]
Hi Guillaume, had you been able to find a workaround ?
I would like to do the same, but during the new user onboarding process, and when creating a profile I don't see another way...
I think that at least if we can't include the self hosted server in the adress , it would be nice when unselectonning Flux and SimpleX servers to be able add self hosted option. Without keeping Flux/SimplexX SMP I can go further

**Comment by ltguillaume** [comment]
Nope, I hope this will be revisited by @epoberezkin some time, because it's something that severely impedes the amount of control the user has over SimpleX.

**Comment by hippi777** [comment]
how configs are managed? if it is layered for purposes with a common fallback mechanism, and possibly with a cache (which can be managed on updating or flagged for lazy-reevaluation), then moving settings between these layers are very simple...

(im still at the beginning of studying simplex and learning haskell... :D )
